kernel/generic/src/mm/as.c

```diff
@@ -487,33 +482,30 @@ NO_TRACE static uintptr_t as_get_unmapped_area(as_t *as, uintptr_t bound,
 	}
 
 	/* Eventually check the addresses behind each area */
-	list_foreach(as->as_area_btree.leaf_list, leaf_link, btree_node_t, node) {
-
-		for (btree_key_t i = 0; i < node->keys; i++) {
-			as_area_t *area = (as_area_t *) node->value[i];
+	as_area_t *area = as_area_first(as);
+	while (area != NULL) {
+		mutex_lock(&area->lock);
 
-			mutex_lock(&area->lock);
+		addr = ALIGN_UP(area->base + P2SZ(area->pages), PAGE_SIZE);
```

Comment:
area->base + P2SZ(area->pages) is already aligned.

Reply:
I guess if you trust area->base to be aligned, then why not.